feat: add script that prepares the output of run.sh for molgenis-py-consensus

marikaris opened the pull request:
M8632

marikaris committed:
feat: add script that prepares the output of run.sh for molgenis-py-consensus

dennishendriksen commented on data-release-pipeline/prepare_for_molgenis/ArgumentParser.py:
```diff
@@ -0,0 +1,45 @@
+import argparse
```
Why not make this part of the data release bash script instead of introducing new scripts in a different language?

marikaris replied:
Because there are already several scripts written in python and writing bash takes me days, rather than hours. I simply didn't have the time to do that and I think a script is better than a written out explanation about how to do it by hand

dennishendriksen commented on data-release-pipeline/prepare_for_molgenis/main.py:
```diff
@@ -0,0 +1,155 @@
+
+
+def retrieve_molgenis_py_consensus(vkgl_dir):
+    print('Retrieving newest molgenis-py-consensus...', end=' ')
```
`run.sh` works with a fixed version of `data-transform-vkgl` to ensure that results can be reproduced. this should work the same for `molgenis-py-consensus`. otherwise results might not be reproducible and in time older versions of run.sh will fail to run when major changes are made to `molgenis-py-consensus`.

marikaris replied:
In most cases you want to use the most recent version, of all tools. `run.sh` is linked to `data-transform-vkgl` because the code is in the same repository. There it makes more sense to link the versions. `molgenis-py-consensus` however is a complete separate repository. I think removing this automatic step will lead to unnecessary manual work. It also means that whenever you update `molgenis-py-consensus` the `data-transform-vkgl` repository always needs to be updated as well. Most of the time when you alter one of those repositories, you don't need to alter the other. So this is also an additional step in the process that complicates things. I however think that it is necessary to have the option to select a version for reproduction purposes.

dennishendriksen commented on data-release-pipeline/prepare_for_molgenis/setup.py:
```diff
@@ -0,0 +1,14 @@
+from setuptools import setup, find_packages
```
how does this script work on e.g. the gearshift cluster with the PythonPlus module?

marikaris replied:
I think/hope it works out of the box. I'll verify it, and if it doesn't, a virtualenv will fix it. I'll add the required steps that work to the SOP.

marikaris committed:
feat: add molgenis-py-consensus version argument

Remove the versionchecker, it doesn't take long to download and it is a lot of code that isn't fully
waterproof to check if the version is already there, so it's best to say goodbye to it.
